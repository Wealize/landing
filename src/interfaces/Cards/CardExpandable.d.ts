/* eslint-disable camelcase */

export interface CardExpandable {
  headline: string
  subheadline: string
  description: string
  image: React.ReactElement
  backgroundColor: string
  textColor: string
  isExpanded: boolean = false
}
