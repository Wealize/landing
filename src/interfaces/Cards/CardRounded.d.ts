/* eslint-disable camelcase */
export interface CardRounded {
  trans_key?: string
  imageUrl: string
  title: string
  description: string
  link: CardRoundedLink | null
  tags: string[]
  hasPrimary: boolean
}
