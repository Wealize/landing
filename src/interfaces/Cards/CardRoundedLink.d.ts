export interface CardRoundedLink {
  text: string
  url: string
  color: string
}
