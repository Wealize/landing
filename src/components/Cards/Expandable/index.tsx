import React, { useState } from 'react'

import { CardExpandable } from '../../../interfaces/Cards/CardExpandable'
import RoundArrowIcon from '../../icons/thingsWeAreGoodAt/RoundArrowIcon'
import {
  Container,
  Header,
  Content,
  Headline,
  Toggle,
  ImageContainer,
  Subheadline,
  Description
} from './styles'

const ExpandableCard = (props: CardExpandable) => {
  const {
    headline,
    subheadline,
    description,
    image,
    backgroundColor,
    textColor,
    isExpanded
  } = props

  const [isExpandedState, setIsExpandedState] = useState(isExpanded)

  return (
    <Container
      backgroundColor={backgroundColor}
      textColor={textColor}
      isExpanded={isExpandedState}
    >
      <Header isExpanded={isExpandedState}>
        <Headline>{headline}</Headline>
        <Toggle
          textColor={textColor}
          isExpanded={isExpandedState}
          onClick={() => setIsExpandedState(!isExpandedState)}
          aria-label="expand card content button"
        >
          <RoundArrowIcon />
        </Toggle>
      </Header>
      <Content isExpanded={isExpandedState}>
        <ImageContainer>{image}</ImageContainer>
        <Subheadline>{subheadline}</Subheadline>
        <Description>{description}</Description>
      </Content>
    </Container>
  )
}

export default React.memo(ExpandableCard)
