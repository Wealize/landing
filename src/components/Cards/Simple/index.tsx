import React from 'react'

import { Container, Text } from './styles'

interface CardProps {
  children?: string
}

const SimpleCard = ({ children }: CardProps) => {
  return (
    <Container>
      <Text>{children}</Text>
    </Container>
  )
}

export default React.memo(SimpleCard)
