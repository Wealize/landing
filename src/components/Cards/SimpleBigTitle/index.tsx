import React from 'react'

import { Container, Headline, SubHeadline } from './styles'

interface CardProps {
  title: string
  description: string
}

const SimpleBigTitleCard = ({ title, description }: CardProps) => {
  return (
    <Container className="SimpleBigTitleCard">
      <Headline>{title}</Headline>
      <SubHeadline>{description}</SubHeadline>
    </Container>
  )
}

export default React.memo(SimpleBigTitleCard)
