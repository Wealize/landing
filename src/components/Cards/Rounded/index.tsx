import React from 'react'
import Link from 'next/link'
import { BackgroundImage } from 'react-image-and-background-image-fade'

import OpenUrlIcon from '../../icons/OpenUrlIcon'
import { CardRounded } from '../../../interfaces/Cards/CardRounded'

import {
  Container,
  BackgroundContainer,
  ContentContainer,
  Content,
  Title,
  Description,
  LinkContainer,
  LinkAnchor,
  TagsContainer,
  Tag
} from './styles'

const RoundedCard = (props: CardRounded) => {
  const { imageUrl, title, description, link, tags, hasPrimary } = props

  const hasTags = () => tags && tags.length > 0

  return (
    <Container className="rounded-card" hasPrimary={hasPrimary}>
      <BackgroundContainer>
        <BackgroundImage
          width="100%"
          height="100%"
          src={imageUrl}
          lazyLoad={true}
          style={{
            minWidth: '300px',
            minHeight: '290px',
            backgroundSize: 'cover',
            backgroundRepeat: 'no-repeat',
            backgroundPosition: 'top center'
          }}
        ></BackgroundImage>
      </BackgroundContainer>
      <ContentContainer hasPrimary={hasPrimary}>
        <Content>
          <Title>{title}</Title>
          <Description>{description}</Description>
          {link
            ? (
            <LinkContainer color={link.color}>
              <Link href={link.url} passHref>
                <LinkAnchor
                  color={link.color}
                  target="_blank"
                  rel="noreferrer"
                  aria-label={link.text}
                >
                  {link.text}
                </LinkAnchor>
              </Link>
              <OpenUrlIcon color={link.color}></OpenUrlIcon>
            </LinkContainer>
              )
            : null}
        </Content>

        {hasTags()
          ? (
          <TagsContainer>
            {tags.map((tag: string, index: number) => (
              <Tag hasPrimary={hasPrimary} key={index}>
                {tag}
              </Tag>
            ))}
          </TagsContainer>
            )
          : null}
      </ContentContainer>
    </Container>
  )
}

export default React.memo(RoundedCard)
