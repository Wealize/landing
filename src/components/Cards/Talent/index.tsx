import React from 'react'
import Link from 'next/link'
import Image from 'next/image'

import { CardTalent } from '../../../interfaces/Cards/CardTalent'
import { SocialNetwork } from '../../../interfaces/SocialNetwork'

import {
  Container,
  ImageContainer,
  ContentContainer,
  Headline,
  Subheadline,
  SocialNetworksContainer
} from './styles'

const TalentCard = ({
  imageUrl,
  name,
  description,
  socialNetworks
}: CardTalent) => {
  return (
    <Container className="talent-card">
      <ImageContainer>
        <Image src={imageUrl} layout="fill" aria-label={`${name} image`} />
      </ImageContainer>
      <ContentContainer>
        <Headline>{name}</Headline>
        <Subheadline>{description}</Subheadline>
        <SocialNetworksContainer>
          {socialNetworks
            ? socialNetworks.map(
                (socialNetwork: SocialNetwork, index: number) => (
                  <Link href={socialNetwork.url} key={index} passHref>
                    <a target="_blank">{socialNetwork.name}</a>
                  </Link>
                )
              )
            : null}
        </SocialNetworksContainer>
      </ContentContainer>
    </Container>
  )
}

export default React.memo(TalentCard)
