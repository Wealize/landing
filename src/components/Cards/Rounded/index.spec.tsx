import React from 'react'
import { mount } from 'enzyme'

import { CardRounded } from '../../../interfaces/Cards/CardRounded'

import RoundedCard from './index'
import {
  Container,
  BackgroundContainer,
  ContentContainer,
  Content,
  Title,
  Description,
  LinkContainer,
  LinkAnchor,
  TagsContainer,
  Tag
} from './styles'

describe('<RoundedCard />', () => {
  const cardProps: CardRounded = {
    imageUrl: '/img/projects/maldita.jpg',
    title: 'Maldita',
    description:
      'Deployment of a WhatsApp Business Chatbot integrated with the Spanish fact-checking platform focused on combating the spread of disinformation in order to expand their service’s reach',
    link: {
      text: 'hola',
      url: 'url',
      color: '#6666ff'
    },
    tags: ['Media', 'Fact-checking', 'Chatbots'],
    hasPrimary: false
  }
  const wrapperWithAllProps = mount(<RoundedCard {...cardProps} />)

  it('wrapper: should render', () => {
    expect(wrapperWithAllProps).toBeDefined()
  })

  it('wrapperWithAllProps: should show correct components content', () => {
    expect(wrapperWithAllProps.find(Container)).toHaveLength(1)
    expect(wrapperWithAllProps.find(BackgroundContainer)).toHaveLength(1)
    expect(wrapperWithAllProps.find(ContentContainer)).toHaveLength(1)
    expect(wrapperWithAllProps.find(Content)).toHaveLength(1)
    expect(wrapperWithAllProps.find(Title)).toHaveLength(1)
    expect(wrapperWithAllProps.find(Title).text()).toEqual(cardProps.title)
    expect(wrapperWithAllProps.find(Description)).toHaveLength(1)
    expect(wrapperWithAllProps.find(Description).text()).toEqual(
      cardProps.description
    )
    expect(wrapperWithAllProps.find(LinkContainer)).toHaveLength(1)
    expect(wrapperWithAllProps.find(LinkAnchor)).toHaveLength(1)
    expect(wrapperWithAllProps.find(LinkAnchor).text()).toEqual(
      cardProps.link.text
    )
    expect(wrapperWithAllProps.find(TagsContainer)).toHaveLength(1)
    expect(wrapperWithAllProps.find(Tag)).toHaveLength(3)
    expect(wrapperWithAllProps.find(Tag).at(0).text()).toEqual(
      cardProps.tags[0]
    )
    expect(wrapperWithAllProps.find(Tag).at(1).text()).toEqual(
      cardProps.tags[1]
    )
    expect(wrapperWithAllProps.find(Tag).at(2).text()).toEqual(
      cardProps.tags[2]
    )
  })

  it('wrapperWithoutTagsProps: should show correct components content without tags content', () => {
    const cardProps: CardRounded = {
      imageUrl: '/img/projects/maldita.jpg',
      title: 'Maldita',
      description:
        'Deployment of a WhatsApp Business Chatbot integrated with the Spanish fact-checking platform focused on combating the spread of disinformation in order to expand their service’s reach',
      link: {
        text: 'hola',
        url: 'url',
        color: '#6666ff'
      },
      tags: [],
      hasPrimary: false
    }
    const wrapperWithoutTagsProps = mount(<RoundedCard {...cardProps} />)

    expect(wrapperWithoutTagsProps.find(Container)).toHaveLength(1)
    expect(wrapperWithoutTagsProps.find(BackgroundContainer)).toHaveLength(1)
    expect(wrapperWithoutTagsProps.find(ContentContainer)).toHaveLength(1)
    expect(wrapperWithoutTagsProps.find(Content)).toHaveLength(1)
    expect(wrapperWithoutTagsProps.find(Title)).toHaveLength(1)
    expect(wrapperWithoutTagsProps.find(Title).text()).toEqual(cardProps.title)
    expect(wrapperWithoutTagsProps.find(Description)).toHaveLength(1)
    expect(wrapperWithoutTagsProps.find(Description).text()).toEqual(
      cardProps.description
    )
    expect(wrapperWithoutTagsProps.find(LinkContainer)).toHaveLength(1)
    expect(wrapperWithoutTagsProps.find(LinkAnchor)).toHaveLength(1)
    expect(wrapperWithoutTagsProps.find(LinkAnchor).text()).toEqual(
      cardProps.link.text
    )
    expect(wrapperWithoutTagsProps.find(TagsContainer)).toHaveLength(0)
    expect(wrapperWithoutTagsProps.find(Tag)).toHaveLength(0)
  })

  it('wrapperWithoutLinkProps: should show correct components content without link content', () => {
    const cardProps: CardRounded = {
      imageUrl: '/img/projects/maldita.jpg',
      title: 'Maldita',
      description:
        'Deployment of a WhatsApp Business Chatbot integrated with the Spanish fact-checking platform focused on combating the spread of disinformation in order to expand their service’s reach',
      link: null,
      tags: null,
      hasPrimary: false
    }
    const wrapperWithoutLinkProps = mount(<RoundedCard {...cardProps} />)

    expect(wrapperWithoutLinkProps.find(Container)).toHaveLength(1)
    expect(wrapperWithoutLinkProps.find(BackgroundContainer)).toHaveLength(1)
    expect(wrapperWithoutLinkProps.find(ContentContainer)).toHaveLength(1)
    expect(wrapperWithoutLinkProps.find(Content)).toHaveLength(1)
    expect(wrapperWithoutLinkProps.find(Title)).toHaveLength(1)
    expect(wrapperWithoutLinkProps.find(Title).text()).toEqual(cardProps.title)
    expect(wrapperWithoutLinkProps.find(Description)).toHaveLength(1)
    expect(wrapperWithoutLinkProps.find(Description).text()).toEqual(
      cardProps.description
    )
    expect(wrapperWithoutLinkProps.find(LinkContainer)).toHaveLength(0)
    expect(wrapperWithoutLinkProps.find(LinkAnchor)).toHaveLength(0)
    expect(wrapperWithoutLinkProps.find(TagsContainer)).toHaveLength(0)
    expect(wrapperWithoutLinkProps.find(Tag)).toHaveLength(0)
  })
})
