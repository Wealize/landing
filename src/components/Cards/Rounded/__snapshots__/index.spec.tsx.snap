// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoundedCard /> wrapper: should render 1`] = `
<Memo(RoundedCard)
  description="Deployment of a WhatsApp Business Chatbot integrated with the Spanish fact-checking platform focused on combating the spread of disinformation in order to expand their service’s reach"
  hasPrimary={false}
  imageUrl="/img/projects/maldita.jpg"
  link={
    Object {
      "color": "#6666ff",
      "text": "hola",
      "url": "url",
    }
  }
  tags={
    Array [
      "Media",
      "Fact-checking",
      "Chatbots",
    ]
  }
  title="Maldita"
>
  <styles__Container
    className="RoundedCard"
    hasPrimary={false}
  >
    <article
      className="styles__Container-sc-1ukl2m-0 kXnjOF RoundedCard"
    >
      <styles__BackgroundContainer>
        <div
          className="styles__BackgroundContainer-sc-1ukl2m-1 iqFCmf"
        >
          <BackgroundImage
            element="div"
            height="100%"
            lazyLoad={true}
            src="/img/projects/maldita.jpg"
            style={
              Object {
                "backgroundPosition": "top center",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "cover",
                "minHeight": "290px",
                "minWidth": "300px",
              }
            }
            transitionTime="0.3s"
            width="100%"
          >
            <ImageLoader
              lazyLoad={true}
              src="/img/projects/maldita.jpg"
              transitionTime="0.3s"
            >
              <VisibilitySensor
                active={true}
                containment={null}
                delayedCall={false}
                intervalCheck={true}
                intervalDelay={100}
                minTopValue={0}
                offset={Object {}}
                onChange={[Function]}
                partialVisibility={true}
                resizeCheck={false}
                resizeDelay={250}
                resizeThrottle={-1}
                scrollCheck={false}
                scrollDelay={250}
                scrollThrottle={-1}
              >
                <styled.div
                  height="100%"
                  width="100%"
                >
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "",
                          "isStatic": false,
                          "lastClassName": "YfvHE",
                          "rules": Array [
                            "
  ",
                            [Function],
                            ";
  ",
                            [Function],
                            ";
  ",
                            [Function],
                            ";
  position: relative;
  overflow: hidden;
",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                    height="100%"
                    width="100%"
                  >
                    <div
                      className="YfvHE"
                      height="100%"
                      width="100%"
                    >
                      <div
                        style={
                          Object {
                            "backgroundPosition": "top center",
                            "backgroundRepeat": "no-repeat",
                            "backgroundSize": "cover",
                            "height": "100%",
                            "minHeight": "290px",
                            "minWidth": "300px",
                            "width": "100%",
                          }
                        }
                      />
                      <Loader
                        hasLoaded={false}
                        isOnTop={true}
                        transitionTime="0.3s"
                      >
                        <styled.div
                          hasLoaded={false}
                          isOnTop={true}
                          transitionTime="0.3s"
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "",
                                  "isStatic": false,
                                  "lastClassName": "iVGwIn",
                                  "rules": Array [
                                    "
  width: 100%;
  height: 100%;
  background-color: #eee;
  position: absolute;
  top: 0;
  left: 0;
  z-index: ",
                                    [Function],
                                    ";
  opacity: ",
                                    [Function],
                                    ";
  transition: opacity ",
                                    [Function],
                                    " ease;

  &:before {
    content: \\"\\";
    position: absolute;
    top: 0;
    left: 0;
    width: 120%;
    height: 120%;
    transform: rotate(-30deg);
    background-image: linear-gradient(
      100deg,
      rgba(255, 255, 255, 0),
      rgba(255, 255, 255, 0.2) 50%,
      rgba(255, 255, 255, 0) 80%
    );
    background-repeat: no-repeat;
    background-position: 0 0;
    background-size: 100px 100%;
    animation: ",
                                    Keyframes {
                                      "id": "sc-keyframes-ksXkoA",
                                      "inject": [Function],
                                      "name": "ksXkoA",
                                      "rules": Array [
                                        "@-webkit-keyframes ksXkoA{from{background-position:0 0;}to{background-position:calc(100% + 50px);}}",
                                        "@keyframes ksXkoA{from{background-position:0 0;}to{background-position:calc(100% + 50px);}}",
                                      ],
                                      "toString": [Function],
                                    },
                                    " 1s infinite linear;
  }
",
                                  ],
                                },
                                "displayName": "styled.div",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            hasLoaded={false}
                            isOnTop={true}
                            transitionTime="0.3s"
                          >
                            <div
                              className="iVGwIn"
                            />
                          </StyledComponent>
                        </styled.div>
                      </Loader>
                    </div>
                  </StyledComponent>
                </styled.div>
              </VisibilitySensor>
            </ImageLoader>
          </BackgroundImage>
        </div>
      </styles__BackgroundContainer>
      <styles__ContentContainer
        hasPrimary={false}
      >
        <div
          className="styles__ContentContainer-sc-1ukl2m-2 gHLKfP"
        >
          <styles__Content>
            <div
              className="styles__Content-sc-1ukl2m-3 fFcpoX"
            >
              <styles__Title>
                <h4
                  className="styles__Title-sc-1ukl2m-4 eEjrll"
                >
                  Maldita
                </h4>
              </styles__Title>
              <styles__Description>
                <p
                  className="styles__Description-sc-1ukl2m-5 kPgGTM"
                >
                  Deployment of a WhatsApp Business Chatbot integrated with the Spanish fact-checking platform focused on combating the spread of disinformation in order to expand their service’s reach
                </p>
              </styles__Description>
              <styles__LinkContainer
                color="#6666ff"
              >
                <div
                  className="styles__LinkContainer-sc-1ukl2m-6 dzNsrd"
                  color="#6666ff"
                >
                  <Link
                    href="url"
                    passHref={true}
                  >
                    <styles__LinkAnchor
                      aria-label="hola"
                      color="#6666ff"
                      href="/url"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      rel="noreferrer"
                      target="_blank"
                    >
                      <a
                        aria-label="hola"
                        className="styles__LinkAnchor-sc-1ukl2m-7 jPychA"
                        color="#6666ff"
                        href="/url"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        rel="noreferrer"
                        target="_blank"
                      >
                        hola
                      </a>
                    </styles__LinkAnchor>
                  </Link>
                  <Memo(OpenUrlIcon)
                    color="#6666ff"
                  >
                    <svg
                      color="#6666ff"
                      height=".8em"
                      viewBox="0 0 11.966 12.059"
                      width=".8em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        id="prefix__Grupo_4662"
                        transform="translate(-281.995 -7112.112)"
                      >
                        <g
                          id="prefix__Grupo_4426"
                          transform="translate(282.745 7112.862)"
                        >
                          <path
                            className="prefix__cls-1"
                            d="M6889.425-3355.839h-5.518v10.559h10.466v-5.392"
                            fill="none"
                            id="prefix__Trazado_11470"
                            stroke="#6666ff"
                            strokeWidth={1.5}
                            transform="translate(-6883.908 3355.839)"
                          />
                          <path
                            className="prefix__cls-1"
                            d="M6907.384-3355.839h3.2v3.307"
                            fill="none"
                            id="prefix__Trazado_11471"
                            stroke="#6666ff"
                            strokeWidth={1.5}
                            transform="translate(-6900.186 3355.839)"
                          />
                          <path
                            className="prefix__cls-1"
                            d="M6902.391-3355.839l-6.067 6.306"
                            fill="none"
                            id="prefix__Trazado_11472"
                            stroke="#6666ff"
                            strokeWidth={1.5}
                            transform="translate(-6891.996 3355.839)"
                          />
                        </g>
                      </g>
                    </svg>
                  </Memo(OpenUrlIcon)>
                </div>
              </styles__LinkContainer>
            </div>
          </styles__Content>
          <styles__TagsContainer>
            <div
              className="styles__TagsContainer-sc-1ukl2m-8 lgKwGG"
            >
              <styles__Tag
                hasPrimary={false}
                key="0"
              >
                <span
                  className="styles__Tag-sc-1ukl2m-9 eBUEOf"
                >
                  Media
                </span>
              </styles__Tag>
              <styles__Tag
                hasPrimary={false}
                key="1"
              >
                <span
                  className="styles__Tag-sc-1ukl2m-9 eBUEOf"
                >
                  Fact-checking
                </span>
              </styles__Tag>
              <styles__Tag
                hasPrimary={false}
                key="2"
              >
                <span
                  className="styles__Tag-sc-1ukl2m-9 eBUEOf"
                >
                  Chatbots
                </span>
              </styles__Tag>
            </div>
          </styles__TagsContainer>
        </div>
      </styles__ContentContainer>
    </article>
  </styles__Container>
</Memo(RoundedCard)>
`;
